_posts/2021-10-01-how-to-pre-assignments.md

```diff
@@ -0,0 +1,67 @@
+
+## Make sure that code compiles and tests pass
+Make sure that the instructions on how to run the project actually work. The reviewer checking out your project does not want to fix stuff to make it work. If you have tests, they should all pass or at least you should have some explanation for failing tests. 
+
+## Focus on getting features complete
+Pre-assignment can consist of many requirements and features. Some of them can be required and some optional. For this fall’s Solita Dev Academy pre-assignment we didn't have any must dos and the applicant's could choose what they wanted to focus on.
+
+If you think that you can't finish everything, it's better to do some things thoroughly and leave the others undone than to do everything half-way. Choose your focus points and what you want to showcase, then start with those and do other things afterwards.
+
+## Write good quality code
+Good quality code is easy to read and looks nice. For instance make sure that you always use the same indentation across your project and remove any unused variables and functions. Use linters and code formatters to automatise this process, because otherwise you will miss a lot of stuff. These tools can visually highlight problems in your code and you can enable format on save, which can for instance add missing semicolons and remove extra white space.
+
+When using linters make sure to fix all problems before submitting your project. It doesn't look good if your application runs with a bunch of linter errors in the console.
+
+Try to make your code easily understandable without using any comments: Organise your code in a logical manner both into different files and within a single file. Use descriptive names for your variables and functions, for instance instead of adding a comment that tells what your function does, try to name your functions in such a way that its purpose can be understood from its name.
+
+Also, don't mix English and Finnish (or any other language) in your code. If you want to use some other language than English, then everything but the programming language commands should be in that language.
+
+## Clean up unused code
+In many cases there is a deadline for pre-assignment and you might not have time to finish up everything you have started. You might have some functions and components that you don't use at all, and those should be cleaned up before submitting. Unused code distracts the reviewer and the general feeling of the codebase could seem unclear. If you are using any linter tools, they will often show error messages for things such as unused imports and variables.
+
+If you have used some boilerplate as base for the project, remember to also clean up unused boilerplate. 
+
+It's of course okay to leave some unused code to your project, but before submitting go through your code. Consider adding a comment if you end up leaving some unused code to your project to tell the reviewer that it was done on purpose.
+
+## To conclude
```

Comment:
Seems like this would be great opportunity to drop a few links, for example Solita open positions and/or dev academy, or even Solita Youtube playlist containing some best practices, for example https://www.youtube.com/watch?v=DNwGS3CzMis&list=PLsaEf6-Yla5qGs0eLovx4neTGgOFvQQ6x or https://www.youtube.com/watch?v=bYeJ3xfwjjg&list=PLsaEf6-Yla5oOMz04xXQ-NsVnYqPCcmZH